### ISSUE TITLE
OpenSSL binaries installed for 1.0.2d are reporting as version 0.9.8 

### ISSUE THREAD

**Issue by Redth** [Homebrew/homebrew#45424]
I'm not sure if the libraries are actually stale, but if you look at the openssl bottles for yosemite and el-capitan, version 1.0.2d, the actual `libcrypto.dylib` library's call to `SSLeay` is returning a version `0x009081df` which corresponds to v0.9.8, not 1.0.2d.

Could someone please check this out? Thanks!

**Comment by DomT4** [comment]
Just to double check the obvious here, are you calling just 'libcrypto' without any LDFLAG or $PATH modification? If so you'll need to pass an LDFLAG or force-link OpenSSL (not recommended) for a plain call to libcrypto to find the right one instead of the systerm's.

**Comment by DomT4** [comment]
*system's. Editing this on my phone so the typo will have to stay in the original reply :)

**Comment by Redth** [comment]
To complicate things a bit, I'm actually pinvoking the .dylib's from C# code (so I've copied the homebrew 1.0.2d's libcrypto.1.0.0.dylib and libssl.1.0.0.dylib locally and pinvoking them.  If I build openssl 1.0.2d from source and pinvoke them the same way, the expected correct version is reported.

tl;dr I do believe i'm actually loading the dylib's from homebrew, not somewhere else on my system

**Comment by Redth** [comment]
Hmm, digging a bit further perhaps it's related to architectures.... I'm going to close this for now, will reopen if I can find anything more concrete..  Thanks
